homeassistant/components/blebox/climate.py

```diff
@@ -40,23 +53,31 @@ class BleBoxClimateEntity(BleBoxEntity[blebox_uniapi.climate.Climate], ClimateEn
     """Representation of a BleBox climate feature (saunaBox)."""
 
     _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
-    _attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT]
     _attr_temperature_unit = TEMP_CELSIUS
 
+    @property
+    def hvac_modes(self):
+        """Return list of supported HVAC modes."""
+        return [HVACMode.OFF, self.hvac_mode]
+
     @property
     def hvac_mode(self):
         """Return the desired HVAC mode."""
         if self._feature.is_on is None:
             return None
-
+        if self._feature.mode is not None:
+            return BLEBOX_TO_HVACMode[self._feature.mode]
         return HVACMode.HEAT if self._feature.is_on else HVACMode.OFF
 
     @property
     def hvac_action(self):
         """Return the actual current HVAC action."""
+        if self._feature.is_on is False and self._feature.hvac_action is not None:
+            return HVACAction.OFF
```

Comment:
The (complete logic) feels overly complicated, shouldn't it be OFF when the device is off? Basically:
1. OFF if the device is off. Or when does it happen that the action is set, but the device itself is off?
2. If `hvac_action` is not `None`, read from `BLEBOX_TO_HVACACTION`. I assume this does not report the heating correctly as the fallback is needed?
3. Fallback  to heating or idle.